app/code/Magento/InventorySales/Model/IsProductSalableCondition/IsSalableLegacyStockItemIsInStock.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Model\IsProductSalableCondition;
+
+use Magento\InventorySalesApi\Api\IsProductSalableInterface;
+use Magento\CatalogInventory\Api\StockConfigurationInterface;
+use Magento\CatalogInventory\Model\Spi\StockRegistryProviderInterface;
```

Comment:
dependency on CatalogInventory in the new code is prohibited. 
we have a dedicated interface  `\Magento\InventoryConfigurationApi\Api\GetStockItemConfigurationInterface::execute`
you should rely on it